### ISSUE TITLE
Process hangs

### ISSUE THREAD

**Issue by michaelmerrill** [fauna-labs/fauna-schema-migrate#15]
It looks like the process is hanging on the recent version of this package. The command is successful, but never exits.

```bash
$ node_modules/.bin/fauna-schema-migrate < state | apply | rollback >
```

package version: `2.0.0` 
node version: `v12.18.0`
mac version: `11.2.3`

**Comment by fauna-brecht** [comment]
Thanks for reporting, will look into that. I thought I fixed that but noticed the same yesterday.

**Comment by chronark** [comment]
It took quite some time for me but it finished eventually.

```

 █▀▀ ▄▀█ █ █ █▄ █ ▄▀█ 
 █▀  █▀█ █▄█ █ ▀█ █▀█   Schema Migrate 2.0.0
 ──────────────────────────────────────────────────
 Executing command: Apply unapplied migrations against the database

    ✔ Retrieved current migration state
      ╭────────────────────────────────────────╮
      │2021-03-12T11:01:23.040Z  ← cloud state │
      │2021-03-12T11:05:20.050Z  ← apply target│
      ╰────────────────────────────────────────╯
    ✔ [ DB: ROOT ] Generated migration code
      ╭────────────────────────────────────────────────────────────────────────────────────────────╮
      │Let([{                                                                                      │
      │  var0: Select(["ref"], CreateIndex({                                                       │
      │    name: "users_by_email",                                                                 │
      │    source: Collection("users"),                                                            │
      │    terms: [{                                                                               │
      │      field: ["data", "email"]                                                              │
      │    }],                                                                                     │
      │    unique: true,                                                                           │
      │    serialized: true                                                                        │
      │  }))                                                                                       │
      │}], Map(["2021-03-12T11:05:20.050Z"], Lambda("migration", Create(Collection("migrations"), {│
      │  data: {                                                                                   │
      │    migration: Var("migration")                                                             │
      │  }                                                                                         │
      │}))))                                                                                       │
      ╰────────────────────────────────────────────────────────────────────────────────────────────╯
    ✔ Done applying migrations

──────────────────────────────────────────────────

Done in 146.58s.
```

**Comment by chronark** [comment]
Might be due to the 60s timeout to ensure the cache is clear?

**Comment by fauna-brecht** [comment]
Introduced by http2 sessions in the fauna driver it's a TLSSocket that remains open. Default timeout is 60.
Will be solved (or is already solved apparently in our driver but still needs a release). New timeout will be 0.5s, later we'll add an optional parameter for the driver.

**Comment by fauna-brecht** [comment]
Should be fixed when paired with the 4.1.2 fauna driver

**Comment by brunosmm** [comment]
Hi @fauna-brecht,

I just update to 4.1.3 fauna driver, did you notice if the hangs reappear on this version? 
It's hanging again for me... I went from 4.1.1 to 4.1.3.

**Comment by fauna-brecht** [comment]
There was a revert of that timeout since it introduced undesired behavior.

**Comment by rts-rob** [comment]
This should have been (re-)fixed with the [release of v4.2.0 of the JavaScript driver](https://github.com/fauna/faunadb-js/releases/tag/4.2.0).

Marking as closed; thank you for reporting and please re-open if you run into this again!
